ion-schema/src/ion_extension.rs

```diff
@@ -0,0 +1,45 @@
+use ion_rs::element::{Element, Value};
+use ion_rs::external::bigdecimal::BigDecimal;
+use ion_rs::{Decimal, Int};
+use num_traits::ToPrimitive;
+use std::str::FromStr;
+
+/// Trait for adding extensions to [`Element`] that are useful for implementing Ion Schema.
+pub(crate) trait ElementExtensions {
+    /// Returns the value as an `usize` if it is an Ion Int that can be represented as such.
+    fn as_usize(&self) -> Option<usize>;
+    /// Returns the value as an `u64` if it is an Ion Int that can be represented as such.
+    fn as_u64(&self) -> Option<u64>;
+    /// Returns the value as a [`Decimal`] if it is any numeric Ion value that can be represented as a [`Decimal`].
+    fn any_number_as_decimal(&self) -> Option<Decimal>;
+}
+impl ElementExtensions for Element {
+    fn as_usize(&self) -> Option<usize> {
+        match self.value() {
+            Value::Int(Int::I64(i)) => i.to_usize(),
+            Value::Int(Int::BigInt(i)) => i.to_usize(),
+            _ => None,
+        }
+    }
```

Comment:
> Non-blocking concern: for these conversions, note that it will not be possible to tell the difference between "The Element was not an integer" and "The Element was an integer that couldn't fit in a usize."

I think that scenario is unlikely to be a problem—it will only occur if people are using numbers that are too large for `u64` and `usize` as arguments to constraints in their ISL. We can revisit it once the `expect_TYPE()` functions are available. For now, I'll make the documentation a little more explicit.